puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/rpm_packaging.groovy

```diff
@@ -53,6 +56,63 @@ pipeline {
             }
         }
 
+        stage("Verify version and release"){
+            when {
+                expression { packages_to_build }
+            }
+            steps {
+                script {
+
+                    for(int i = 0; i < packages_to_build.size(); i++) {
+                        def index = i
+                        packages[packages_to_build[index]] = {
+                            package_name = packages_to_build[index]
+
+                            new_version = query_rpmspec("packages/**/${package_name}/*.spec", '%{VERSION}')
+                            new_release = query_rpmspec("packages/**/${package_name}/*.spec", '%{RELEASE}')
+
+                            sh "git checkout origin/${env.ghprbTargetBranch}"
+
+                            old_version = query_rpmspec("packages/**/${package_name}/*.spec", '%{VERSION}')
+                            old_release = query_rpmspec("packages/**/${package_name}/*.spec", '%{RELEASE}')
+
+                            sh "git checkout -"
+
+                            compare_version = sh(
+                              script: "rpmdev-vercmp ${old_version} ${new_version}",
+                              returnStatus: true
+                            )
+
+                            compare_release = sh(
+                              script: "rpmdev-vercmp ${old_release} ${new_release}",
+                              returnStatus: true
+                            )
+
+                            compare_new_to_one = sh(
+                              script: "rpmdev-vercmp 1 ${new_release}",
+                              returnStatus: true
+                            )
+
+                            if (compare_version != VERCMP_EQUAL && (compare_new_to_one == VERCMP_OLDER || compare_new_to_one == VERCMP_EQUAL)) {
```

Comment:
The `compare_new_to_one == VERCMP_OLDER` is to catch the release is set to 0.1?

Reply:
I believe so.